cranelift/codegen/src/machinst/lower.rs

```diff
@@ -961,10 +1041,16 @@ impl<'func, I: VCodeInst> LowerCtx for Lower<'func, I> {
         self.inst_constants.get(&ir_inst).cloned()
     }
 
-    fn get_input(&self, ir_inst: Inst, idx: usize) -> LowerInput {
+    fn maybe_get_input_as_source_or_const(&self, ir_inst: Inst, idx: usize) -> LowerInput {
```

Comment:
Returning `Option` would also make it so the naming could be a little more natural here, since the "maybe" would become redundant, and this could be named `fn get_input_as_non_reg(...) -> Option<NonRegInput> {...}` which I think reads a lot better.

Reply:
I decided to remove `maybe_` as the returned struct is composed of `Option`s, so it's semantically pretty clear what's going on in any case.